conans/client/hook_manager.py

```diff
@@ -10,8 +10,13 @@
 from conans.util.files import save
 
 attribute_checker_hook = """
+from conans.model.conan_file import is_alias_conanfile
```

Comment:
We have to be careful with the imports to the conan internals in hooks as any change on this will brake the hook. In This case is less important as this hook is embedded in the client

Reply:
Maybe use the `getattr()` instead to avoid this as done in the lines below?

Reply:
I created the `is_alias_conanfile` to do the same check everywhere. Maybe this check should be a classmethod:

```python
class ConanFile:
    ...
    @classmethod
    def is_alias_conanfile(cls):
        return hasattr(cls, 'alias')

    ...
```

but I don't want to be the first one introducing a non-functional method in `ConanFile` class.

----

I feel like `alias` is going to evolve, I don't know how, people wants the `deprecation` notice there, maybe declaring overrides only to `user` `channel` but not the full alias string.... That's the reason I wanted to have it centralized.

Reply:
I agree with @danimtb that it would be better to avoid a full abs path ``from conans.model.conan_file``, which would be necessary if we want to import this functionality, even as a class method. I think in the hook it might be better to do the ``getattr()`` for now.